### PR TITLE
Sound Refactor (Take II)

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Sound Refactor, take II.

Refactors sound a bit (not as much as before).

- Adds in support for playing local sounds from atoms
- Fixes a runtime related to lobby preference toggles.
- cleans up the code just a bit

### CODE REVIEW

File: code/game/sound.dm [hunk 1]
```diff
@@ -105,16 +96,16 @@ var/const/FALLOFF_SOUNDS = 0.5
...
-			if ("rustle") soundin = pick(rustle_sound)
-			if ("bodyfall") soundin = pick(bodyfall_sound)
-			if ("punch") soundin = pick(punch_sound)
-			if ("clownstep") soundin = pick(clown_sound)
-			if ("jackboot") soundin = pick(jackboot_sound)
-			if ("swing_hit") soundin = pick(swing_hit_sound)
-			if ("hiss") soundin = pick(hiss_sound)
-			if ("pageturn") soundin = pick(page_sound)
-			//if ("gunshot") soundin = pick(gun_sound)
+			if ("shatter") soundin = pick('sound/effects/Glassbr1.ogg','sound/effects/Glassbr2.ogg','sound/effects/Glassbr3.ogg')
```
Comment (tigercat2000) [inline code/game/sound.dm]: i still don't get why this is any better than the list, it's less readable

File: code/game/sound.dm [hunk 2]
```diff
@@ -37,8 +24,7 @@ var/list/page_sound = list('sound/effects/pageturn1.ogg', 'sound/effects/pagetur
 
 var/const/FALLOFF_SOUNDS = 0.5
 
-/mob/proc/playsound_local(var/turf/turf_source, soundin, vol as num, vary, frequency, falloff, is_global)
-	if(!src.client || ear_deaf > 0)	return
+/atom/proc/playsound_local(var/turf/turf_source, soundin, vol as num, vary, frequency, falloff, is_global)
```
Comment (tigercat2000) [inline code/game/sound.dm]: But.. that's not what this does, it doesn't play sounds, it sends sounds to whatever src is.

**Comment by tigercat2000** [comment]
>Adds in support for playing local sounds from atoms

You mean what playsound is literally for, playing a sound to every mob around an atom in a certain range?

**Comment by Spacemanspark** [comment]
RIP.

**Pull request closed without merging**